src/internal/middleware.js

```diff
@@ -37,20 +37,24 @@ export default function sagaMiddlewareFactory(options = {}) {
   function sagaMiddleware({getState, dispatch}) {
     runSagaDynamically = runSaga
     const sagaEmitter = emitter()
+    const sagaDispatch = !sagaMonitor ? dispatch : wrapSagaDispatch(dispatch)
```

Comment:
hmm... dont think so. for #598 seems the external dispatch needs to be queued as well (which is not the case actually) but I think it's more related to the asap#isSuspended state. More fundamentally IMO what we need primarily is to come up with some unambiguous semantics for scheduling before tempting any impl.